include/npcomp/Dialect/Torch/IR/TorchOps.td

```diff
@@ -190,13 +190,17 @@ def Torch_MethodOp : Torch_Op<"method", [
     method `name` which calls `function`. `function` is an unbound function.
     That is, it explicitly takes the torch.nn.Module as a parameter (no implicit
     "self" object).
+
+    If `private` is present, it indicates that external calls cannot be made
+    to this method.
   }];
 
-  let arguments = (ins StrAttr:$name, FlatSymbolRefAttr:$function);
+  // `private` is a C++ keyword, so use `isPrivate`.
+  let arguments = (ins StrAttr:$name, FlatSymbolRefAttr:$function, UnitAttr:$isPrivate);
   let results = (outs);
 
   let assemblyFormat = [{
-    $name `,` $function attr-dict
+    (`private` $isPrivate^)? $name `,` $function attr-dict
```

Comment:
Why not match visibility attributes that functions use?

Reply:
torch.method and torch.attr are not Symbol's, and thus sym_visibility doesn't make sense for them. (also if you look at what sym_visibility provides, e.g. the concept of "nested", it doesn't really make sense in this context).